server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotsServiceIT.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.snapshots;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.elasticsearch.action.ActionFuture;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.snapshots.mockstore.MockRepository;
+import org.elasticsearch.test.MockLogAppender;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+
+public class SnapshotsServiceIT extends AbstractSnapshotIntegTestCase {
+
+    public void testDeletingSnapshotsIsLoggedAfterClusterStateIsProcessed() throws InterruptedException {
+        createRepository("test-repo", "fs");
+        createIndexWithRandomDocs("test-index", randomIntBetween(1, 42));
+        createSnapshot("test-repo", "test-snapshot", List.of("test-index"));
+
+        final MockLogAppender mockLogAppender = new MockLogAppender();
+
+        try {
+            mockLogAppender.start();
+            Loggers.addAppender(LogManager.getLogger(SnapshotsService.class), mockLogAppender);
+
+            mockLogAppender.addExpectation(
+                new MockLogAppender.UnseenEventExpectation(
+                    "[does-not-exist]",
+                    SnapshotsService.class.getName(),
+                    Level.INFO,
+                    "deleting snapshots [does-not-exist] from repository [test-repo]"
+                )
+            );
+
+            mockLogAppender.addExpectation(
+                new MockLogAppender.SeenEventExpectation(
+                    "[deleting test-snapshot]",
+                    SnapshotsService.class.getName(),
+                    Level.INFO,
+                    "deleting snapshots [test-snapshot] from repository [test-repo]"
+                )
+            );
+
+            mockLogAppender.addExpectation(
+                new MockLogAppender.SeenEventExpectation(
+                    "[test-snapshot deleted]",
+                    SnapshotsService.class.getName(),
+                    Level.INFO,
+                    "snapshots [test-snapshot/*] deleted"
```

Comment:
I think this is going to be a little flaky because we aren't waiting for the deletion to complete yet. A call to `awaitNoMoreRunningOperations()` before `assertAllExpectationsMatched()` should do the trick tho.

Reply:
I forgot about the background deletion process. Thanks for catching it.